permission/permission.go

```
@@ -0,0 +1,759 @@
+
+// Monitors account access related events and updates the cache accordingly
+func (p *PermissionCtrl) manageAccountPermissions() {
+	chAccessModified := make(chan *pbind.AcctManagerAccountAccessModified)
+	chAccessRevoked := make(chan *pbind.AcctManagerAccountAccessRevoked)
+	chStatusChanged := make(chan *pbind.AcctManagerAccountStatusChanged)
+
+	var evtAccessModified *pbind.AcctManagerAccountAccessModified
+	var evtAccessRevoked *pbind.AcctManagerAccountAccessRevoked
+	var evtStatusChanged *pbind.AcctManagerAccountStatusChanged
+
+	opts := &bind.WatchOpts{}
+	var blockNumber uint64 = 1
+	opts.Start = &blockNumber
+
+	if _, err := p.permAcct.AcctManagerFilterer.WatchAccountAccessModified(opts, chAccessModified); err != nil {
+		log.Info("Failed NewNodeProposed", "error", err)
+	}
+
+	if _, err := p.permAcct.AcctManagerFilterer.WatchAccountAccessRevoked(opts, chAccessRevoked); err != nil {
+		log.Info("Failed NewNodeProposed", "error", err)
+	}
+
+	if _, err := p.permAcct.AcctManagerFilterer.WatchAccountStatusChanged(opts, chStatusChanged); err != nil {
+		log.Info("Failed NewNodeProposed", "error", err)
+	}
+
+	for {
+		select {
+		case evtAccessModified = <-chAccessModified:
+			types.AcctInfoMap.UpsertAccount(evtAccessModified.OrgId, evtAccessModified.RoleId, evtAccessModified.Address, evtAccessModified.OrgAdmin, types.AcctStatus(int(evtAccessModified.Status.Uint64())))
+
+		case evtAccessRevoked = <-chAccessRevoked:
+			types.AcctInfoMap.UpsertAccount(evtAccessRevoked.OrgId, evtAccessRevoked.RoleId, evtAccessRevoked.Address, evtAccessRevoked.OrgAdmin, types.AcctActive)
+
+		case evtStatusChanged = <-chStatusChanged:
+			ac := types.AcctInfoMap.GetAccount(evtStatusChanged.Address)
+			types.AcctInfoMap.UpsertAccount(evtStatusChanged.OrgId, ac.RoleId, evtStatusChanged.Address, ac.IsOrgAdmin, types.AcctStatus(int(evtStatusChanged.Status.Uint64())))
+		}
+	}
+}
+
+// Disconnect the node from the network
+func (p *PermissionCtrl) disconnectNode(enodeId string) {
```

error is swallowed
better to capture and return an error when disconnecting node

---

have logged the error. This is running in a separate go routing and hence preferred logging of the error.